src/pretix/control/forms/page.py

```diff
@@ -0,0 +1,68 @@
+from urllib.request import urlopen
+
+import lxml.html
+from django import forms
+from django.core.files.base import ContentFile
+from django.core.files.storage import default_storage
+from django.utils.crypto import get_random_string
+from django.utils.translation import gettext_lazy as _
+
+from pretix.base.models.page import Page
+
+
+class PageSettingsForm(forms.ModelForm):
+    class Meta:
+        model = Page
+        fields = (
+            "title",
+            "slug",
+            "link_on_website_start_page",
+            "link_in_header",
+            "link_in_footer",
+            "require_confirmation",
+            "text"
+        )
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields['link_on_website_start_page'].widget = forms.HiddenInput()
+
+    mimes = {
+        "image/gif": "gif",
+        "image/jpeg": "jpg",
+        "image/png": "png",
+        "image/webp": "webp",
+    }
+
+    def clean_slug(self):
+        slug = self.cleaned_data["slug"]
+        if Page.objects.filter(slug=slug).exists():
+            raise forms.ValidationError(
+                _("You already have a page on that URL."),
+                code="duplicate_slug",
+            )
+        return slug
+
+    def clean_text(self):
+        t = self.cleaned_data["text"]
+        for locale, html in t.data.items():
```

Comment:
This code is too long, move it to an utility function and call it in `clean_text`.
Btw, because this code is to create and save a file to our system, the file will be come garbage if at the end, this model data is invalid and no model is saved.

So, please move the code which saves file to the `save()` method, because at that point, we are sure that the data is valid and ready to create a model instance.